File: core-api-layer/southville-nhs-school-portal-api-layer/src/users/dto/create-admin.dto.ts
import {
  IsString,
  IsOptional,
  MinLength,
  MaxLength,
  Matches,
  IsDateString,
} from 'class-validator';
import { ApiProperty } from '@nestjs/swagger';
import { CreateUserDto, UserRole, UserType } from './create-user.dto';

export class CreateAdminDto extends CreateUserDto {
  @IsDateString()
  @ApiProperty({
    example: '1980-03-20',
    description: 'Birthday',
    type: 'string',
    format: 'date',
  })
  birthday: string;

  @IsOptional()
  @IsString()
  @MaxLength(255)
  @ApiProperty({
    example: 'System Administrator',
    description: 'Role description',
    required: false,
    maxLength: 255,
  })
  roleDescription?: string;

  @IsOptional()
  @IsString()
  @MinLength(10)
  @MaxLength(15)
  @Matches(/^\+?[1-9]\d{1,14}$/, {
    message: 'Phone number must be a valid international format',
  })
  @ApiProperty({
    example: '+1234567890',
    description: 'Phone number in international format',
    required: false,
    minLength: 10,
    maxLength: 15,
  })
  phoneNumber?: string;

  // Set default values for base class
  constructor() {
    super();
    this.role = UserRole.ADMIN;
    this.userType = UserType.ADMIN;
  }
}

File: desktop-app/Southville8BEdgeUI/Models/Api/CreateAdminDto.cs
using System.Text.Json.Serialization;

namespace Southville8BEdgeUI.Models.Api;

public class CreateAdminDto
{
    [JsonPropertyName("email")]
    public string Email { get; set; } = string.Empty;

    [JsonPropertyName("fullName")]
    public string FullName { get; set; } = string.Empty;

    [JsonPropertyName("role")]
    public string Role { get; set; } = "Admin";

    [JsonPropertyName("userType")]
    public string UserType { get; set; } = "admin";

    [JsonPropertyName("birthday")]
    public string Birthday { get; set; } = string.Empty;

    [JsonPropertyName("roleDescription")]
    public string? RoleDescription { get; set; }

    [JsonPropertyName("phoneNumber")]
    public string? PhoneNumber { get; set; }
}
